### ISSUE TITLE
Warning: Cannot read property 'stage' of undefined

### ISSUE THREAD

**Issue by AndriusA** [assemble/assemble-middleware-rss#8]
Just trying to use the middleware for the first time and the error puzzles me. I am getting the same error when trying to use with my site, so also tried running the tests of the middleware. After installing dependencies and running grunt from the package directory it fails with:

```
Running "clean:actual" (clean) task
>> 0 paths cleaned.

Running "assemble:test-1" (assemble) task
Warning: Cannot read property 'stage' of undefined Use --force to continue.

Aborted due to warnings.
```

Any hints as to why it might be?

**Comment by jonschlinkert** [comment]
could you put the contents of your gruntfile in a gist and link it here? might be easier to debug.

**Comment by AndriusA** [comment]
was running it from the directory where npm installs it in the project, so that's the Gruntfile.js that's provided here (https://github.com/assemble/assemble-middleware-rss/blob/master/Gruntfile.js). This morning getting the same warning, but about a different property ('format').

Just tried to run it from a freshly cloned repo:

```
 git clone https://github.com/assemble/assemble-middleware-rss.git
 cd assemble-middleware-rss/
 npm install
 grunt
```

Now getting a different error, but nevertheless an error:

```
Running "clean:actual" (clean) task

Running "assemble:test-1" (assemble) task
Assembling test/dest/bar.html OK
Assembling test/dest/baz.html OK
Assembling test/dest/foo.html OK
>> 3 pages assembled.

Running "assemble:test-2" (assemble) task
Assembling test/dest/bar.html OK
Assembling test/dest/baz.html OK
Assembling test/dest/foo.html OK
>> 3 pages assembled.

Running "assemble:test-3" (assemble) task
Assembling test/dest/bar.html OK
Assembling test/dest/bar.html ERROR
Warning: Cannot read property 'format' of undefined Use --force to continue.

Aborted due to warnings.
```

In the previous case it was (and just verified it again, although now getting the above error about the 'format' property):

```
npm install assemble-middleware-rss --save-dev
cd node_modules/assemble-middleware-rss/
npm install
grunt
```

(output as above)

The blog part of my own Gruntfile is:

```
blog: {
    options: {
      layout: './template/blogPostLayout.html',
      plugins: ['assemble-middleware-rss'],
      rss: {
        logging: true,
        author: "Andrius Aucinas",
        dest: "./static/feed.xml",
        siteurl: "http://exaple.com/",
        title: 'Title',
        description: 'Descr',
        format: 'true'
      }
    },
    files: {
      './static/blog.html': ['./template/blog.html'],
      './static/blogPosts/': ['./blog/*.md'],
    }
  }
```

And executing it:

```
Running "assemble:blog" (assemble) task
Assembling static/blog.html OK
Assembling static/blog.html ERROR
Warning: Cannot read property 'format' of undefined Use --force to continue.
```

**Comment by jonschlinkert** [comment]
I think this is the cause of the `format` error: https://github.com/assemble/assemble-middleware-rss/blob/4719ad53b1f0af5249425d6309f19ed96e6b0458/rss.js#L127. we'll need to add the following or its equivalent:

``` js
options = options || {};
options.feed = options.feed || {};
```

and the logic in the `options.feed.format` comparison needs a little work.

not sure about the `stage` error yet

**Comment by AndriusA** [comment]
somehow stage has gone away now... And with your pointer it has managed to build a feed xml.

The option in my gruntfile was also wrong, it has to be feed rather than rss (please update the docs).

Perhaps worth creating a new issue, but the 'dest' option does not seem to work either. Keeping it as above ('dest: "./static/feed.xml"'), the file gets generated in the middleware's own directory:

```
find . -name feed.xml
./node_modules/assemble-middleware-rss/feed.xml
```

**Comment by jonschlinkert** [comment]
thanks for the detail! very helpful.

**Comment by pburtchaell** [comment]
Sorry for the delayed response @AndriusA, but I have been on vacation and haven't had the best internet for the past week. Most of the issues you had should be fixed in [v0.2.2](https://github.com/assemble/assemble-middleware-rss/releases).

> The option in my gruntfile was also wrong, it has to be feed rather than rss (please update the docs).

We switched between `rss` and `feed` twice for name of the middleware's config object. At some point in the development, the documentation didn't get updated properly. The correct usage is `rss`. 

This is fixed in https://github.com/assemble/assemble-middleware-rss/commit/43f7562a2b447b111141ff63112c9042a2f19994 and https://github.com/assemble/assemble-middleware-rss/commit/fa89917aa1dec213bbb54a844dede2759327cd92

> The file gets generated in the middleware's own directory

This has been fixed https://github.com/assemble/assemble-middleware-rss/commit/8b28ecb4a3418eb62764f6afa215a4376da76702. By default, the file will be generated in the same directory of your pages (`page.dirname`), but the `dest` option can also be used to change the directory.

> The logic in the options.feed.format comparison needs a little work - @jonschlinkert 

I think most of these errors were caused by the `rss` vs `feed` issue, if you still see improvements that could be made, let me know.
